pkg/spoke/controllers/statuscontroller/availablestatus_controller.go

```diff
@@ -0,0 +1,265 @@
+				available += 1
+			case metav1.ConditionFalse:
+				unavailable += 1
+			case metav1.ConditionUnknown:
+				unknown += 1
+			}
+		}
+	}
+
+	switch {
+	case unavailable > 0:
+		return workapiv1.StatusCondition{
+			Type:    string(workapiv1.WorkAvailable),
+			Status:  metav1.ConditionFalse,
+			Reason:  "ResourcesNotAvailable",
+			Message: fmt.Sprintf("%d of %d resources are not available", unavailable, len(manifests)),
+		}
+	case unknown > 0:
+		return workapiv1.StatusCondition{
+			Type:    string(workapiv1.WorkAvailable),
+			Status:  metav1.ConditionUnknown,
+			Reason:  "ResourcesStatusUnknown",
+			Message: fmt.Sprintf("%d of %d resources have unknown status", unknown, len(manifests)),
+		}
+	default:
+		return workapiv1.StatusCondition{
+			Type:    string(workapiv1.WorkAvailable),
+			Status:  metav1.ConditionTrue,
+			Reason:  "ResourcesAvailable",
+			Message: "All resources are available",
+		}
+	}
+}
+
+// buildAvailableStatusCondition returns a StatusCondition with type Available for a given manifest resource
+func buildAvailableStatusCondition(resourceMeta workapiv1.ManifestResourceMeta, dynamicClient dynamic.Interface) workapiv1.StatusCondition {
+	conditionType := string(workapiv1.ManifestAvailable)
+
+	if len(resourceMeta.Resource) == 0 || len(resourceMeta.Version) == 0 || len(resourceMeta.Name) == 0 {
+		return workapiv1.StatusCondition{
+			Type:    conditionType,
+			Status:  metav1.ConditionUnknown,
+			Reason:  "IncompletedResourceMeta",
+			Message: "Resource meta is incompleted",
+		}
+	}
+
+	available, err := isResourceAvailable(resourceMeta.Namespace, resourceMeta.Name, schema.GroupVersionResource{
+		Group:    resourceMeta.Group,
+		Version:  resourceMeta.Version,
+		Resource: resourceMeta.Resource,
+	}, dynamicClient)
+	if err != nil {
+		return workapiv1.StatusCondition{
+			Type:    conditionType,
+			Status:  metav1.ConditionUnknown,
+			Reason:  "FetchingResourceFailed",
+			Message: fmt.Sprintf("Failed to fetch resource: %v", err),
+		}
+	}
+
+	if available {
+		return workapiv1.StatusCondition{
+			Type:    conditionType,
+			Status:  metav1.ConditionTrue,
+			Reason:  "ResourceAvailable",
+			Message: "Resource is available",
+		}
+	}
+
+	return workapiv1.StatusCondition{
+		Type:    conditionType,
+		Status:  metav1.ConditionFalse,
+		Reason:  "ResourceNotAvailable",
+		Message: "Resource is not available",
+	}
+}
+
+// isResourceAvailable checks if the specific resource is available or not
+func isResourceAvailable(namespace, name string, gvr schema.GroupVersionResource, dynamicClient dynamic.Interface) (bool, error) {
+	_, err := dynamicClient.Resource(gvr).Namespace(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func mergeStatusConditions(conditions []workapiv1.StatusCondition, newCondition workapiv1.StatusCondition) []workapiv1.StatusCondition {
```

Comment:
what is the difference between this and helpser.MergeStatusConditions?

Reply:
helper.MergeStatusConditions(conditions, newConditions) returns A + B, while:
  - A is the intersection of conditions and newConditions, which is merged
  - B is the difference of newConditions and conditions (newConditions - conditions)

And mergeStatusConditions(conditions, newCondition) returns the union of [newCondition] and conditions, while the intersection of these two slices is merged.